django/db/backends/sqlite3/schema.py

```diff
@@ -150,7 +150,9 @@ def is_self_referential(f):
             body.pop(old_field.name, None)
             mapping.pop(old_field.column, None)
             body[new_field.name] = new_field
-            if old_field.null and not new_field.null:
+            if new_field.generated:
+                pass
+            elif old_field.null and not new_field.null:
```

Comment:
This looks like correct though I'm just wondering with this new `pass` branch whether this for block could be written a little better 😅

Would it be better to move the `rename_mapping[old] = new` line up and turn the `pass` into a `continue`